### PR TITLE
Implement FingerprintClientHello to generate ClientHelloSpec from ClientHello raw bytes

### PR DESCRIPTION
**Pull request by max-b**
This is a continuation of https://github.com/refraction-networking/utls/pull/58 from my personal account so that I can get this PR over the finish line.

It hopefully will close https://github.com/refraction-networking/utls/issues/54

@sergeyfrolov I responded to your comments on the previous/getlantern PR, I'm sorry that this is a kind of fragmented solution, but hopefully I've addressed your previous concerns.

Thanks!

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -102,6 +102,26 @@ you can set UConn.HandshakeStateBuilt = true, and marshal clientHello into UConn
...
+generatedSpec, err := fingerprinter.FingerprintClientHello(rawCapturedClientHelloBytes)
+if err != nil {
+  panic("fingerprinting failed: %v", err)
+}
+if err := uConn.ApplyPreset(generatedSpec); err != nil {
+  panic("applying generated spec failed: %v", err)
+}
+```
+The `rawCapturedClientHelloBytes` should be the full tls record, including the record type/version/length header.
+
```
Comment (max-b) [inline README.md:124]: I thought this could be useful, but wanted to flag it for you to make sure you think it's correct/clear/etc.

**Comment by max-b** [comment]
I've removed the `DisableSecretsRegeneration` flag/functionality and added a short piece of documentation. Let me know what you think!

File: README.md (hunk 1 above)
Comment (sergeyfrolov) [reply README.md:124]: Thanks! I was opening this PR with the thought to request a README entry but you already done that.

**Comment by sergeyfrolov** [comment]
Many thanks for your contribution!

**Comment by max-b** [comment]
:tada: Happy to be able to contribute! Thanks again for your patience!

**Pull request merged**